lib/coinbase/webhook.rb

```diff
@@ -0,0 +1,167 @@
+        model = Coinbase.call_api do
+          webhooks_api.create_webhook(
+            create_webhook_request: {
+              network_id: Coinbase.normalize_network(network_id),
+              notification_uri: notification_uri,
+              event_type: event_type,
+              event_filters: event_filters
+            }
+          )
+        end
+
+        new(model)
+      end
+
+      # Enumerates the webhooks.
+      # The result is an enumerator that lazily fetches from the server, and can be iterated over,
+      # converted to an array, etc...
+      # @return [Enumerable<Coinbase::Webhook>] Enumerator that returns webhooks
+      def list
+        Coinbase::Pagination.enumerate(lambda(&method(:fetch_webhooks_page))) do |webhook|
+          Coinbase::Webhook.new(webhook)
+        end
+      end
+
+      private
+
+      def webhooks_api
+        Coinbase::Client::WebhooksApi.new(Coinbase.configuration.api_client)
+      end
+
+      def fetch_webhooks_page(page)
+        webhooks_api.list_webhooks({ limit: DEFAULT_PAGE_LIMIT, page: page })
+      end
+    end
+
+    # Initializes a new Webhook object.
+    #
+    # @param model [Coinbase::Client::Webhook] The underlying Webhook object.
+    # @raise [ArgumentError] If the model is not a Coinbase::Client::Webhook.
+    def initialize(model)
+      raise ArgumentError, 'model must be a Webhook' unless model.is_a?(Coinbase::Client::Webhook)
+
+      @model = model
+    end
+
+    # Returns the ID of the webhook.
+    #
+    # @return [String] The ID of the webhook.
+    def id
+      @model.id
+    end
+
+    # Returns the network ID associated with the webhook.
+    #
+    # @return [String] The network ID of the webhook.
+    def network_id
+      @model.network_id
+    end
+
+    # Returns the notification URI of the webhook.
+    #
+    # @return [String] The URI where notifications are sent.
+    def notification_uri
+      @model.notification_uri
+    end
+
+    # Returns the event type of the webhook.
+    #
+    # @return [String] The type of event the webhook listens for.
+    def event_type
+      @model.event_type
+    end
+
+    # Returns the event filters applied to the webhook.
+    #
+    # @return [Array<Hash>] An array of event filters used by the webhook.
+    def event_filters
+      @model.event_filters
+    end
+
+    # Updates the webhook with a new notification URI.
+    #
+    # @param notification_uri [String] The new URI for webhook notifications.
+    # @return [self] Returns the updated Webhook object.
+    #
+    # @example Update the notification URI of a webhook
+    #   webhook.update(notification_uri: 'https://new-url.com/callback')
+    def update(notification_uri:)
+      serialized_event_filters = event_filters.map do |filter|
+        {
+          contract_address: filter['contract_address'],
+          from_address: filter['from_address'],
+          to_address: filter['to_address']
+        }.compact
```

Comment:
```suggestion
        }
```

`compact` is used to remove `nil` items from an array, however you're building up an array that cannot have any `nil` items (since you're explicitly constructing the hash).

Reply:
any of these can be nil, so using compact make sense, though it probably doesn't matter as backend will only query non-nil

Reply:
Gotcha, you shouldn't need to do any explicit handling like that if we just serialize the event filters using the `WebhookEventFilter` serialization via `to_hash`